src/main/ts/rules/NoImplicitDomGlobals.ts

```diff
@@ -108,7 +94,13 @@ export const noImplicitDomGlobals: Rule.RuleModule = {
   create: (context) => {
     const options = context.options[0] || {};
     const allowed = getAllowedGlobals(options);
-    const invalid = Globals.getDomGlobals().filter((name) => !allowed.includes(name));
+    // PERFORMANCE: Convert the invalid list to an object for better lookup times
+    const invalid = Globals.getDomGlobals()
+      .filter((name) => !allowed.includes(name))
+      .reduce<Record<string, boolean>>((acc, name) => {
```

Comment:
[nitpick, feel free to ignore] this feels less idiomatic than just using a `Set` (which would also avoid all the potential weirdnesses in having prototype methods and data being accessible / modifiable in the same ways).

Reply:
We discussed this last week... at the moment this still works on older NodeJS versions and we'd have to do a breaking change to change this to a `Set`

Reply:
Discussed more on slack and it seems like this package was originally set up to support `ES2017` (see the tsconfig.json settings), however it was missing the minimum nodejs version in package.json. As such I've swapped this over to a `Set` and fixed the missing `engines` config.